### PR TITLE
Add model: CSPDarknet

### PR DESCRIPTION
**Pull request by illian01**
## Description

Please include a summary in English, of the changes in this pull request. If it closes an issue, please mention it here.

Closes: ~

You should link at least one existing issue for PR. Before your create a PR, please check to see if there is an issue for this change.  
PRs from forked repository not accepted.

## Change(s)

- Add CSPDarknet backbone

## Changelog

If you PR to `dev` branch, please add a brief summary of the change to the **Upcoming Release** section of the [`CHANGELOG.md`](https://github.com/Nota-NetsPresso/netspresso-trainer/blob/master/CHANGELOG.md) file and include a link to the PR (formatted in markdown) and a link to your github profile.

For example,

```
- Added a new feature by `@myusername` in [PR 2023](https://github.com/Nota-NetsPresso/netspresso-trainer/pull/2023)
```

## Code Formatting

If you PR to either `master` or `dev` branch, you should follow the code linting process. Please check your code with `lint_check.sh` in `./scripts` directory.
For more information, please read the contribution guide in `CONTRIBUTING.md`.

### CODE REVIEW

File: src/netspresso_trainer/models/op/custom.py [hunk 1]
```diff
@@ -489,3 +489,142 @@ def forward(self, x: Tensor) -> Tensor:
...
+                              out_channels=hidden_channels, 
+                              kernel_size=1, 
+                              stride=1, act_type=act_type)
+        self.conv3 = ConvLayer(in_channels=2 * hidden_channels, 
+                               out_channels=out_channels, 
+                               kernel_size=1, 
+                               stride=1, act_type=act_type)
+        
+        # Newly define because of slight difference with Bottleneck of custom.py
+        class DarknetBlock(nn.Module):
```
Comment (deepkyu) [inline src/netspresso_trainer/models/op/custom.py:559]: @illian01 

Should it be internally defined? It seems like no issue but is quite confusing to others...

File: src/netspresso_trainer/models/op/custom.py [hunk 2]
```diff
@@ -489,3 +489,142 @@ def forward(self, x: Tensor) -> Tensor:
 
     # def __repr__(self):
     #     return "{}(type={})".format(self.__class__.__name__, self.pool_type)
+
+
+class Focus(nn.Module):
```
Comment (deepkyu) [inline src/netspresso_trainer/models/op/custom.py:494]: Self-note:

This would be not compatible with NP Compressor because of reshaping including channel dim.

**Review by deepkyu — COMMENTED** [review]
@illian01 

Thx for your hard work. Always welcome for a new model :) 
Plz check the inline comment

File: src/netspresso_trainer/models/op/custom.py (hunk 2 above)
Comment (illian01) [reply src/netspresso_trainer/models/op/custom.py:494]: I tested CSPDarknet with PyNP, and fortunately, it works well.

File: src/netspresso_trainer/models/op/custom.py (hunk 1 above)
Comment (illian01) [reply src/netspresso_trainer/models/op/custom.py:559]: It doesn't need to be inside of `CSPLayer`. 
I thought it is less confusing ... but that may be because I wrote that code.
I will move `DarknetBlock` to outside.

Comment (deepkyu) [reply src/netspresso_trainer/models/op/custom.py]: @illian01 

Actually, I should not ask about this cause' our ruff check doesn't say anything :)
Sorry for confusion, but it would be better if we get some code style guideline 👍🏻

**Review by deepkyu — APPROVED** [review]
